### PR TITLE
(SERVER-1109) Add semicolon to allowed characters

### PR DESCRIPTION
**Pull request by jpinsonault**
Follow up PR to add semicolons to the allowed list of characters in a code-id

### CODE REVIEW

File: test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj [hunk 1]
```diff
@@ -63,7 +63,9 @@
   (testing "valid-code-id? accepts valid code-ids"
     (is (vc-core/valid-code-id? nil))
     (is (vc-core/valid-code-id? "dcf16ec"))
-    (is (vc-core/valid-code-id? "master-plan:stage_1:destroy-all-humans")))
+    (is (vc-core/valid-code-id? "whatMakesJavaBad;isPartlySemiColons;clojureIsRealNice"))
```
Comment (haus) [inline test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj]: Can you add a test case that is the urn we expect to pass in PE?

File: test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj [hunk 2]
```diff
@@ -63,7 +63,9 @@
...
+    (is (vc-core/valid-code-id? "master-plan:stage_1:destroy-all-humans"))
+    (is (vc-core/valid-code-id? "combining:lots;of_valid-characters")))
```
Comment (KevinCorcoran) [inline test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj]: @jpinsonault wanna add one more example to this - a URN example like Josh posted in his comment on the previous PR?

**Comment by jpinsonault** [comment]
Added a PE style code-id test case

**Comment by KevinCorcoran** [comment]
:+1:

File: test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj [hunk 3]
```diff
@@ -63,7 +63,10 @@
   (testing "valid-code-id? accepts valid code-ids"
     (is (vc-core/valid-code-id? nil))
     (is (vc-core/valid-code-id? "dcf16ec"))
-    (is (vc-core/valid-code-id? "master-plan:stage_1:destroy-all-humans")))
+    (is (vc-core/valid-code-id? "whatMakesJavaBad;isPartlySemiColons;clojureIsRealNice"))
+    (is (vc-core/valid-code-id? "master-plan:stage_1:destroy-all-humans"))
+    (is (vc-core/valid-code-id? "combining:lots;of_valid-characters"))
+    (is (vc-core/valid-code-id? "urn:code-id:1:4dcf1fd;production")))
```
Comment (haus) [inline test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj]: Nitpick. The PE style code id would be: `urn:puppet:code-id:1:4dcf1fd;production`.

Comment (haus) [inline test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj]: but this is probably close enough for our needs.

**Pull request merged**